src/server/gc_worker/gc_manager.rs

```diff
@@ -442,13 +446,23 @@ impl<S: GcSafePointProvider, R: RegionInfoProvider + 'static, E: KvEngine> GcMan
         let mut progress = Some(Key::from_encoded(BEGIN_KEY.to_vec()));
 
         // Records how many region we have GC-ed.
-        let mut processed_regions = 0;
+        let mut scheduled_regions = 0;
+        let task_controller = Arc::new((Mutex::new(0), Condvar::new()));
+        let (lock, cvar) = &*task_controller;
+        let maybe_wait = |max_tasks| {
+            let mut current_tasks: std::sync::MutexGuard<'_, usize> = lock.lock().unwrap();
+            while *current_tasks > max_tasks {
+                // Wait until the number of current tasks is below the limit
+                current_tasks = cvar.wait(current_tasks).unwrap();
```

Comment:
ok, this part looks good.

Reply:
Intuitively, I thought this piece of code runs in multiple threads and they race to get a chance to run once a slot is available. But, after reading the code, seems to me, this does not run concurrently. `CondVar` here is only used as a way to communicate with the worker? From the previous comments, seems there was a discussion about this already. I am fine with the current implementation, but maybe it is better to leave a comment to describe the usage (not the use of `current task`, but the use of `condvar`) for clarification?

Reply:
ok